### PR TITLE
feat: device inventory data aggregation

### PR DESCRIPTION
**Pull request by tranchitella**
Ticket: MEN-5916
Changelog: Title

Signed-off-by: Fabio Tranchitella <fabio.tranchitella@northern.tech>

### CODE REVIEW

File: docs/management_api.yml [hunk 1]
```diff
@@ -182,6 +210,47 @@ components:
...
 
+    AggregationTerm:
+      type: object
+      properties:
+        name:
+          type: string
+          description: Name of the aggregation.
+        attribute:
+          type: string
+          description: Attribute key to aggregate.
```
Comment (mzedel) [inline docs/management_api.yml:221]: this should also reflect the ability to do partial matches on attributes (see e.g. the example in the ticket)

**Review by mzedel — COMMENTED** [review]
the general direction is spot on, but especially the software information needs the aggregation on partial matches - so I just want to be explicit that this is not lost...

File: docs/management_api.yml [hunk 2]
```diff
@@ -182,6 +210,47 @@ components:
...
+        scope:
+          type: string
+          description: The scope the attribute exists in.
+        size:
+          type: integer
+          description: Field to run the aggregation for.
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply docs/management_api.yml:227]: max number of records to return, defaults to 0 (all the results).
This is part of the Opensearch/Elasticsearch implementation, if you want to get up to the N most popular results.
I can remove it, if you want.

File: docs/management_api.yml (hunk 1 above)
Comment (tranchitella) [reply docs/management_api.yml:221]: Ok, let me work on it.

File: docs/management_api.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mzedel) [reply docs/management_api.yml:227]: aah ok, yes that makes sense... is then maybe `limit` more fitting?

[the remark this answers is not in the document]
Comment (tranchitella) [reply docs/management_api.yml:227]: yup, I'll rename it as `limit`; `size` is the Elasticsearch/Opensearch term, but we don't want to base our APIs on their jargon.

**Comment by tranchitella** [comment]
@mender-test-bot start pipeline --fast

File: docs/management_api.yml [hunk 3]
```diff
@@ -182,6 +225,59 @@ components:
...
 
+    AggregationTerm:
+      type: object
+      properties:
+        name:
+          type: string
+          description: Name of the aggregation.
+        attribute:
+          type: string
+          description: Attribute key(s) to aggregate; it supports wildcards using the `*` character.
```
Comment (kjaskiewiczz) [inline docs/management_api.yml:236]: does it support wildcards?

File: docs/management_api.yml [hunk 4]
```diff
@@ -182,6 +225,59 @@ components:
...
+          type: string
+          description: The scope the attribute(s) exists in.
+        limit:
+          type: integer
+          description: Number of top results to return.
+          default: 10
+        aggregations:
+          type: array
+          items:
+            $ref: '#/components/schemas/AggregationTerm'
```
Comment (kjaskiewiczz) [inline docs/management_api.yml:247]: should we limit number of aggregation terms? or it doesn't make sense?

**Review by kjaskiewiczz — COMMENTED** [review]
looking at the code it seems fine, but at the same time I'm lacking expertise here, so I'm not approving yet; I'm planning to play with it tomorrow before approving

File: docs/management_api.yml (hunk 3 above)
Comment (tranchitella) [reply docs/management_api.yml:236]: Oh, not anymore; that was the original plan, but we decided to skip this feature.

File: docs/management_api.yml (hunk 4 above)
Comment (tranchitella) [reply docs/management_api.yml:247]: Yes, it makes sense. I will add some (high) limits just to be on the safe side.

File: api/http/management.go [hunk 5]
```diff
@@ -62,6 +62,55 @@ type attribute struct {
...
+			err,
+		)
+		return
+	}
+
+	c.JSON(http.StatusOK, res)
+}
+
+func parseAggregateParams(ctx context.Context, c *gin.Context) (*model.AggregateParams, error) {
+	var aggrgateParams model.AggregateParams
```
Comment (kjaskiewiczz) [inline api/http/management.go:90]:
```suggestion
	var aggregateParams model.AggregateParams
```

File: docs/management_api.yml [hunk 6]
```diff
@@ -13,6 +13,49 @@ security:
...
+      responses:
+        200:
+          description: OK. Returns a list of aggregations.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Aggregation'
+              example:
```
Comment (kjaskiewiczz) [inline docs/management_api.yml:47]: the result is 
```
{
  "name": "some-name",
  "items": [
      {
        "key": "foo",
        "count": 1
      }
  ],
  "other_count":5
}
  ```
so the example is wrong

**Review by kjaskiewiczz — APPROVED** [review]
minor comments only; tested locally and it works

**Pull request merged**